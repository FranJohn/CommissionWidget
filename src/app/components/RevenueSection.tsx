import React from 'react';
import SubWidget from './SubWidget';
import InputField from './InputFeild';

/**
 * Props for the RevenueSection component.
 */
interface RevenueSectionProps {
    handleRevenueChange: (value: number | null) => void;
    handleCalculateCommission: () => void;
}

/**
 * A component to represent the revenue section with input and calculate button.
 * @param {RevenueSectionProps} props - Properties for the RevenueSection component.
 * @returns {JSX.Element} - JSX for the RevenueSection component.
 */
const RevenueSection: React.FC<RevenueSectionProps> = ({ handleRevenueChange, handleCalculateCommission }) => {
    return (
        <SubWidget title={["Calculate Commission"]}>
            <InputField
                placeholder="Enter revenue value..."
                onChange={handleRevenueChange}
            />   
            <button onClick={handleCalculateCommission} data-testid="calculate-button">Calculate</button>
        </SubWidget>
    );
};

export default RevenueSection;
