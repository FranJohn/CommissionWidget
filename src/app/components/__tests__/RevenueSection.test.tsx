import '@testing-library/jest-dom';
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import RevenueSection from '../RevenueSection';

describe('RevenueSection component', () => {
    it('renders input field and button', () => {
        const handleRevenueChange = jest.fn();
        const handleCalculateCommission = jest.fn();

        const { getByPlaceholderText, getByText } = render(
            <RevenueSection
                handleRevenueChange={handleRevenueChange}
                handleCalculateCommission={handleCalculateCommission}
            />
        );

        const inputElement = getByPlaceholderText('Enter revenue value...');
        expect(inputElement).toBeInTheDocument();

        const calculateButton = getByText('Calculate');
        expect(calculateButton).toBeInTheDocument();
    });

    it('calls provided functions on input change and button click', () => {
        const handleRevenueChange = jest.fn();
        const handleCalculateCommission = jest.fn();

        const { getByPlaceholderText, getByText } = render(
            <RevenueSection
                handleRevenueChange={handleRevenueChange}
                handleCalculateCommission={handleCalculateCommission}
            />
        );

        const inputElement = getByPlaceholderText('Enter revenue value...');
        fireEvent.change(inputElement, { target: { value: '500' } });
        expect(handleRevenueChange).toHaveBeenCalledWith(500);

        const calculateButton = getByText('Calculate');
        fireEvent.click(calculateButton);
        expect(handleCalculateCommission).toHaveBeenCalled();
    });

    it('matches the SubWidget snapshot', () => {
        const handleRevenueChange = jest.fn();
        const handleCalculateCommission = jest.fn();

        const { container } = render(
            <RevenueSection
                handleRevenueChange={handleRevenueChange}
                handleCalculateCommission={handleCalculateCommission}
            />
        );       
        expect(container).toMatchSnapshot();
    });
});
