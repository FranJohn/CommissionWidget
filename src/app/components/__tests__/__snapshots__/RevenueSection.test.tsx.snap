// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RevenueSection component matches the SubWidget snapshot 1`] = `
<div>
  <div
    class="sub-widget"
  >
    <h5>
      Calculate Commission
    </h5>
    <div
      class="sub-widget-content"
    >
      <div
        class="input-wrapper"
      >
        <input
          class="custom-input"
          placeholder="Enter revenue value..."
          type="text"
        />
        GBP
      </div>
      <button
        data-testid="calculate-button"
      >
        Calculate
      </button>
    </div>
  </div>
</div>
`;
